File: main/sherlock-and-gcd/sherlock-and-gcd.pl
#!/usr/bin/env perl

use strict;
use warnings;
use List::Util qw(reduce);

sub gcd {
    my ($m, $n) = @_;
    return $n == 0 ? $m : gcd($n, $m % $n);
}

sub gcd_list {
    my @nums = @_;
    return reduce { gcd $a, $b } sort { $a <=> $b } @nums;
}

sub read_record {
    return split /\s+/msx, <>;
}

for (1..<>) {
    <>; # don't need n
    print gcd_list(read_record) == 1 ? "YES\n" : "NO\n";
}

File: main/sherlock-and-gcd/sherlock-and-gcd.py
#!/usr/bin/env python3

from fractions import gcd
from functools import reduce

def read_record():
    return map(int, input().split())

def gcd_list(nums):
    return reduce(gcd, sorted(nums))

for _ in range(int(input())):
    input() # don't need n
    print('YES' if gcd_list(read_record()) == 1 else 'NO')
